opm/autodiff/FullyImplicitBlackoilSolver.hpp

```diff
@@ -194,6 +194,12 @@ namespace Opm {
         std::vector<int>         primalVariable_;
 
         // Private methods.
+
+        // return true if wells are available
+        bool wellsActive() const { return wells_ ? wells_->number_of_wells > 0 : false ; }
+        // return wells object
+        const Wells& wells () const { assert( wells_ ); return *wells_; }
```

Comment:
> Could you make an example?

I don't think I understand what you mean.  The input to macro `assert` **must** be a boolean and the pointer expression `wells_` is not a boolean.  Said expression is **convertible** to boolean, but `assert` does not enforce a boolean context although most typical implementations do.

So, either write `assert (wells_ != 0)` or circumvent the entire issue by writing
```C++
assert (wellsActive())
```